syntax = "proto3";

import "google/protobuf/any.proto";
import "enums.proto";

package pslx;

// the next will be 6
message ProtoTable {
    string table_name = 1;
    string table_path = 2;
    string created_time = 3;
    map<string, google.protobuf.Any> data = 4;
    string updated_time = 5;
}

// the next will be 4
message ProtoTableIndexMap {
    map<string, uint64> index_map = 1;
    uint64 cur_shard = 2;
    uint64 size_per_shard  = 3;
}

// the next will be 13
message ContainerBackendValue {
    string container_name = 1;

    message OperatorInfo {
        repeated string parents = 2;
        string start_time = 3;
        string end_time = 4;
        Status status = 5;
        string log_file = 6;
    }

    map<string, OperatorInfo> operator_info_map = 2;

    Status container_status = 3;
    ModeType mode = 4;
    DataModelType data_model = 5;
    string updated_time = 6;
    string start_time = 7;
    string end_time = 8;
    string log_file = 9;
    map<string, int32> counters = 10;
    int32 ttl = 11;
    string run_cell = 12;
}
